libraries/Wire/src/Wire.cpp

```diff
@@ -27,6 +27,7 @@ extern "C" {
 
 #include "Wire.h"
 
+#define I2C_NO_OPTION_FRAME       0xFFFF0000U /*!< XferOptions default value */
```

Comment:
What is this value and how would that be interpreted by ? 
Isn't there a #define amongst I2C_FIRST_AND_LAST_FRAME, I2C_LAST_FRAME,  that can be used instead ?

Reply:
https://github.com/stm32duino/Arduino_Core_STM32/blob/8134a78a09de5b4f6f041c967e2cfaa2b373c210/system/Drivers/STM32F4xx_HAL_Driver/Src/stm32f4xx_hal_i2c.c#L85-L121

I guess I2C_LAST_FRAME_NO_STOP would be better in this case?

Reply:
Hi @fpistm, @LMESTM, I'm back !
Well, `I2C_NO_OPTION_FRAME` comes from the stm32f4xx_hal_i2c.c, line 327.
I agree that it's pretty annoying that there's a lot of #define everywhere...
The goal is to not change the behavior of i2c driver when the repeated start is not used. Therefore, the previous routine `HAL_I2C_Master_Transmit_IT` initializes `hi2c->XferOptions = I2C_NO_OPTION_FRAME` (stm32f4xx_hal_i2c.c, line1694).

So, as this value is not defined in the header file stm32f4xx_hal_i2c.h, I need to re-define it in Wire.cpp.

Regarding your suggestions, it may works maybe. But, honestly, I have made a lot of tries before finding a working configuration. The critical test is in stm32f4xx_hal_i2c.c line 5023:
`if ((CurrentXferOptions != I2C_FIRST_AND_LAST_FRAME) && (CurrentXferOptions != I2C_LAST_FRAME) && (CurrentXferOptions != I2C_NO_OPTION_FRAME))`    
So, I agree that this value would work. But why does I2C_NO_OPTION_FRAME has been defined by MCD application team ? Why this value is outside the header instead of being a part of transfer frame option list ? It's a bit messy.

Unfortunately, I don't have the MMA accelerometer anymore.
I hope that someone who gets a device requiring a repeated-start feature will bring its help.
In parallel, I will do my best to get it again.

Reply:
Hi @jmchiappa, @LMESTM,
@ABOSTM has discussed with our I2C expert and `I2C_NO_OPTION_FRAME` should not be used.
@ABOSTM will provide more info soon.